src/internal/NeoPixelAnimator.cpp

```diff
@@ -164,3 +165,48 @@ void NeoPixelAnimator::UpdateAnimations()
         }
     }
 }
+
+void NeoPixelAnimator::setDuration(uint16_t indexAnimation, uint16_t newDuration)
+{
+    if(indexAnimation >= _countAnimations) { return; } //invalid animation index
+    else 
+    {
+        // get the current animation progress
+        float current_progress = getProgress(indexAnimation);
+
+        // change the duration
+        AnimationContext* pAnim;
+        pAnim = &_animations[indexAnimation];
+        pAnim->_duration = newDuration;
+
+        // _remaining time must also be reset after a duration change; use the original progress to make this change
+        setProgress(indexAnimation, current_progress);
+    }
+}
+
+// actually sets the _remaining ticks left for an animation in its AnimationContext using the progress provided here
+void NeoPixelAnimator::setProgress(uint16_t indexAnimation, float newProgress)
```

Comment:
when not exposed publicly, then setProgress/getProgress become very thin (no parameter checks, and animation already known) so they can be collapsed into the change duration method.